tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4

```diff
@@ -33,22 +33,22 @@ ifdef(`MONO', `',
 define(matrix1, `ROUTE_MATRIX(3,
 			     `BITS_TO_BYTE(1, 0, 0 ,0 ,0 ,0 ,0 ,0)',
 			     `BITS_TO_BYTE(0, 1, 0 ,0 ,0 ,0 ,0 ,0)',
-			     `BITS_TO_BYTE(0, 0, 1 ,0 ,0 ,0 ,0 ,0)',
-			     `BITS_TO_BYTE(0, 0, 0 ,1 ,0 ,0 ,0 ,0)',
-			     `BITS_TO_BYTE(0, 0, 0 ,0 ,1 ,0 ,0 ,0)',
-			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,1 ,0 ,0)',
-			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,1 ,0)',
-			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,1)')')
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,0)')')
 
 define(matrix2, `ROUTE_MATRIX(4,
 			     `BITS_TO_BYTE(1, 0, 0 ,0 ,0 ,0 ,0 ,0)',
 			     `BITS_TO_BYTE(0, 1, 0 ,0 ,0 ,0 ,0 ,0)',
-			     `BITS_TO_BYTE(0, 0, 1 ,0 ,0 ,0 ,0 ,0)',
-			     `BITS_TO_BYTE(0, 0, 0 ,1 ,0 ,0 ,0 ,0)',
-			     `BITS_TO_BYTE(0, 0, 0 ,0 ,1 ,0 ,0 ,0)',
-			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,1 ,0 ,0)',
-			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,1 ,0)',
-			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,1)')')
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,0)')')
```

Comment:
@juimonen can you remind me why we have an 8x8 matrix when a 2x2 seemed to do the job? Is this because we hard-coded the demux component to 8ch?

Reply:
@plbossart we support max 8ch streams, so this is the configuration block for "possibly" all channels. We don't have extra parameters to specify the size or row numbers etc.

I think we had a misunderstanding in the "simplification" code, in a way that demux interpreted all 1's to be used somehow, even though that would lie in a channel count above the stream channel count. So runtime stream channel count should dictate what is used from the matrix, and that is what is now fixed in #3583 

It is like you would have multiband eq, where the top band lies above 8khz, if you send there 4kHz sine this band should not have any effect on your sine. It should not however stop you defining the bands in your config.

Reply:
> @plbossart we support max 8ch streams, so this is the configuration block for "possibly" all channels. We don't have extra parameters to specify the size or row numbers etc.
> 
> I think we had a misunderstanding in the "simplification" code, in a way that demux interpreted all 1's to be used somehow, even though that would lie in a channel count above the stream channel count. So runtime stream channel count should dictate what is used from the matrix, and that is what is now fixed in #3583
> 
> It is like you would have multiband eq, where the top band lies above 8khz, if you send there 4kHz sine this band should not have any effect on your sine. It should not however stop you defining the bands in your config.

Right, but usually you can define the number of bands... I find it odd that we force the demux to deal with 8ch even though we don't have a use case for it on playback. Anyways, that will be for another day and another optimization, this is good enough for now.